lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -338,9 +356,9 @@ namespace Js
             library->AddMember(prototype, aliasFunctionIdentifier, scriptFunction);
         }
 
-        if (functionIdentifier == PropertyIds::values)
+        if (prototype == library->arrayPrototype)
         {
-            library->arrayPrototypeDefaultValuesFunction = func;
+            RecordDefaultIteratorFunctions(functionIdentifier, scriptContext, func);
```

Comment:
what happens if the .values function deleted or overridden? do we see the updated behavior?

Reply:
If values function is deleted or overridden we will see updated behavior. For for-ofs we use Symbol.iterator, we will continue to work the default way for for-ofs.